src/GovUk.Education.ExploreEducationStatistics.Common/Services/ImportStatusService.cs

```diff
@@ -61,98 +58,99 @@ public class ImportStatusService : IImportStatusService
-                    $"Update: {origDataFileName} {statusBefore} ({percentageCompleteBefore}%) -> {status} ({percentageCompleteAfter}%) ignored");
+                    $"Update: {dataFileName} {currentImportStatus.Status} ({percentageCompleteBefore}%) -> " +
+                    $"{newStatus} ({percentageCompleteAfter}%) ignored as this import is already finished");
+                return;
+            }
+
+            // Ignore updating if already aborting and the new state is not aborting or finishing
+            if (currentImportStatus.IsAborting() && !ImportStatus.IsFinishedOrAbortingState(newStatus))
+            {
+                _logger.LogWarning(
+                    $"Update: {dataFileName} {currentImportStatus.Status} ({percentageCompleteBefore}%) -> " +
+                    $"{newStatus} ({percentageCompleteAfter}%) ignored as this import is already aborting or is finished");
+                return;
+            }
+            
+            // Ignore updates if attempting to downgrade from a normal importing state to a lower normal importing state,
+            // or if the percentage is being set lower or the same as is currently and is the same state
+            if (!ImportStatus.IsFinishedOrAbortingState(newStatus) && 
+                (currentImportStatus.Status.CompareTo(newStatus) > 0 || 
+                 currentImportStatus.Status == newStatus && percentageCompleteBefore > percentageCompleteAfter))
+            {
+                _logger.LogWarning(
+                    $"Update: {dataFileName} {currentImportStatus.Status} ({percentageCompleteBefore}%) -> " +
+                    $"{newStatus} ({percentageCompleteAfter}%) ignored");
                 return;
             }
 
             // Ignore updating to an equal percentage complete (after rounding) at the same status without logging it
-            if (statusBefore == status && percentageCompleteBefore == percentageCompleteAfter)
+            if (currentImportStatus.Status == newStatus && percentageCompleteBefore == percentageCompleteAfter)
             {
                 return;
             }
 
             _logger.LogInformation(
-                $"Update: {origDataFileName} {statusBefore} ({percentageCompleteBefore}%) -> {status} ({percentageCompleteAfter}%)");
+                $"Update: {dataFileName} {currentImportStatus.Status} ({percentageCompleteBefore}%) -> {newStatus} ({percentageCompleteAfter}%)");
 
             import.PercentageComplete = percentageCompleteAfter;
-            import.Status = status;
+            import.Status = newStatus;
 
-            try
+            // If this is a status update that MUST be the "winner" when multiple threads are writing to the same Table
+            // Row in storage, set the If-Match header to "*" (via the ETag) to ensure that this update does not check
+            // for other updates between this entity getting retrieved and getting merged.
+            if (ImportStatus.IsFinishedOrAbortingState(newStatus))
```

Comment:
Cool, I really like this change, and it should remove those warning we see for collisions where the majority of the time we don't care about them. 💯

Reply:
Hopefully!